README.md

```diff
@@ -8,92 +8,420 @@
   }
 }
+```
 
-// stores/togglerStore.js
-const Store = require('./Store')
+The `UserStore` iterates through all registered event listeners whenever a
+state change occurs. All listeners will be called with the updated state as a
+first argument:
 
-const initialState = {
-  on: false
+```js
+class UserStore {
+  // ...
+  setState (state) {
+    this.state = state;
+    for (let listener of this.listeners) {
+      listener.call(null, state);
+    }
+  }
 }
+```
 
-module.exports = new Store(initialState)
+And the `<Profile />` component can now call this method in order to register
+for subsequent state changes. In order to trigger a re-render whenever the
+store gets updated, we copy the store's state onto the component's state:
 
-// actions/togglerActions.js
-const togglerStore = require('../stores/togglerStore')
+```js
+class Profile extends React.Component {
+  componentDidMount () {
+    userStore.addListener((state) => {
+      this.setState(state)
+    });
+    this.setState(userStore.getState())
+  }
+  render () {
+    const {userId} = this.props;
 
-function handleClick(e) {
-  e.preventDefault()
+    // We're now accessing `this.state` instead of `userStore`.
+    const profile = this.state.find((user) => user.id === userId);
 
-  // we could update multiple stores here
-  togglerStore.updateState({ on: !togglerStore.state.on })
+    // ...
+  }
 }
+```
 
-// components/Toggler.js
-const React = require('react')
+And Voila! Our component is now wired up to our store.
+There is just one small (but very important!) issue which we didn't address,
+yet!
 
-const togglerActions = require('../actions/togglerActions')
-const togglerStore = require('../stores/togglerStore')
+What happens when the component is being unmounted? In other words, what
+happens when the user leaves the profile page and goes somewhere else?
 
-module.exports = class Toggler extends React.Component {
-  constructor(props) {
-    super(props)
+Subsequent store updates would trigger a `setState` on an unmounted component!
+Not only is this going to trigger a warning in React, it's also a very common
+source for memory leaks in Flux architectures. Whenever we register components
+on a store, we have to make sure we remove all listeners at the point where the
+component is being unmounted.
 
-    this.handleClick = togglerActions.handleClick.bind(this)
-    this.updateState = this.updateState.bind(this)
+`EventEmitter`s usually expose a `removeListener` method for cleaning up event
+listeners, but it's quite common for Flux stores to simply return a function
+from `addListener` that when called removes the listener from the store. Sounds
+complicated? Let's look at the code!
 
-    this.state = togglerStore.state
+```js
+class UserStore {
+  addListener (listener) {
+    this.listeners.push(listener);
+    const removeListener = () => {
+      this.listeners = this.listeners.filter((l) => listener !== l);
+    };
+    return removeListener;
```

Comment:
Can we return a `function` instead of using an arrow function here? Seems kind of redundant to use an arrow function, especially if it takes an equal number of lines